File: Marcianitos/src/deepspace/EnemyStarShip.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author laura
 */
public class EnemyStarShip {
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Loot loot;
    private Damage damage;
    
    EnemyStarShip(String n, float a, float s, Loot l, Damage d) {
        this.name = n;
        this.ammoPower = a;
        this.shieldPower = s;
        this.loot = l;
        this.damage = d;
    }
    
    EnemyStarShip(EnemyStarShip e) {
        this.name = e.name;
        this.ammoPower = e.ammoPower;
        this.shieldPower = e.shieldPower;
        this.loot = e.loot;
        this.damage = e.damage;
    }
    
    EnemyToUI getUIVersion() {
        return new EnemyToUI(this);
    }
    
    public float fire(){
        throw new UnsupportedOperationException();
    }
    
    public float getAmmoPower(){
        return this.ammoPower;
    }
    
    public Damage getDamage(){
        return this.damage;
    }
    
    public Loot getLoot(){
        return this.loot;
    }
    
    public String getName(){
        return this.name;
    }
    
    public float getShieldPower(){
        return this.shieldPower;
    }
    
    public float protection(){
        throw new UnsupportedOperationException();
    }
    
    public ShotResult receiveShot(float shot){
        throw new UnsupportedOperationException();
    }
}

File: RUBY/src/EnemyStarShip.rb
#encoding: UTF-8

require_relative 'ShotResult'

module Deepspace
	class EnemyStarShip
		# Constructor
		def initialize (n, ap, sp, l, d)  # n string, ap float, sp float, l loot, d Damage
			@name = n
			@ammoPower = ap
			@shieldPower = sp
			@loot = l
			@damage = d
			
		end
		
		# Constructor de copia
		def self.newCopy(other)
			new(other.name, other.ammoPower, other.shieldPower, other.loot, other.damage)
		end
		
		# Consultores publicos
		attr_reader :ammoPower
		attr_reader :loot
		attr_reader :name
		attr_reader :shieldPower
		attr_reader :damage
		
		def protection ()
			return @shieldPower   # devuelve float
		end
		
		def fire ()
			return @ammoPower   # devuelve float
		end
		
		def receiveShot (shot)   # shot es un float
			resultado = ShotResult::RESIST
			
			if (protection() < shot)
				resultado = ShotResult::DONOTRESIST
			end
			
			return (resultado)   # devuelve un ShotResult
		end
		
	end
end
